bench/workgen/workgen.cpp

```diff
@@ -456,63 +486,74 @@ WorkloadRunner::start_tables_drop(WT_CONNECTION *conn)
             continue;
         }
 
-        for (int drops = 0; drops < _workload->options.drop_count; ++drops) {
-            std::vector<std::string> drop_uris;
-            // The data structures for the dynamic table set are protected by a mutex.
-            {
-                const std::lock_guard<std::mutex> lock(*icontext->_dyn_mutex);
-
-                // Walk the list and select the first that is not already in the delete list.
-                for (auto const &it : icontext->_dyn_tint) {
-                    if (icontext->_dyn_table_runtime[it.second]._pending_delete) {
-                        continue;
-                    } else {
-                        VERBOSE(*_workload, "Marking pending removal for: " << it.first);
-                        icontext->_dyn_table_runtime[it.second]._pending_delete = true;
-                        tables_delete.push_back(it.first);
-                        break;
-                    }
-                }
+        std::vector<std::string> drop_files; // Files ready to be dropped.
+        // The data structures for the dynamic table set are protected by a mutex.
+        {
+            const std::lock_guard<std::mutex> lock(*icontext->_dyn_mutex);
 
+            /*
+             * When dropping consider how many dynamic tables we have left, and how many are already
+             * marked for deletion.
+             */
+            int tables_remaining = icontext->_dyn_tint.size() - pending_delete.size();
```

Comment:
Seems like a great place for an assertion here `icontext->_dyn_tint.size() > pending_delete.size()` but I know there is another ticket that will look into adding assertions.

Reply:
Added.
This is a good place for an assert, I have added one. For some reason I thought the remaining tables could go negative and then reset them to 0 - but no that should not happen.